src/devices/bus/mononcol/slot.h

```diff
@@ -0,0 +1,143 @@
+// license:BSD-3-Clause
+// copyright-holders:Fabio Priuli
+#ifndef MAME_BUS_MONONCOL_SLOT_H
+#define MAME_BUS_MONONCOL_SLOT_H
+
+#pragma once
+
+#include "imagedev/cartrom.h"
+
+#include <cassert>
+
+/***************************************************************************
+ TYPE DEFINITIONS
+ ***************************************************************************/
+
+class device_mononcol_cart_interface : public device_interface
+{
+public:
+
+	// construction/destruction
+	virtual ~device_mononcol_cart_interface();
+
+	// reading and writing
+	virtual void rom_alloc(u32 size, int width, endianness_t end, char const *tag);
+
+	u8 *get_rom_base()  { return m_rom; }
+	u32 get_rom_size() { return m_rom_size; }
+
+	virtual uint8_t read() { return 0x00; }
+	virtual DECLARE_WRITE_LINE_MEMBER(dir_w) { }
+	virtual void write(uint8_t data) { };
+	virtual void set_ready() { };
+
+	virtual void set_spi_region(uint8_t* region) { }
+	virtual void set_spi_size(size_t size) { }
```

Comment:
These should all be pure virtual – a concrete cartridge device is going to have to implement them anyway.

Reply:
I think he means = 0;
But I don't really care (at leasty remove the stray ";" here, write and set_ready)

Reply:
set them to pure virtual anyway